src/cli.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{CliArgs, Client, GenesisDetails, StateId};
+use ethereum_consensus::{phase0::mainnet::Fork, primitives::Root};
+use url::Url;
+
+pub async fn cli(args: CliArgs) {
+    //set up client
+    let s = args.node_url;
+    let url: Url = Url::parse(&s).unwrap();
+    let client = Client::new(url);
+    let endpoint = args.endpoint.as_str();
+
+    match endpoint {
+        // Match on the provided endpoint
+        "node_version" => println!("{:?}", client.get_node_version().await.unwrap()),
+        "genesis" => {
+            let out: GenesisDetails = client.get_genesis_details().await.unwrap();
+            println!("{:?}", out.genesis_time);
+            println!("{:?}", out.genesis_fork_version);
+            println!("{:?}", out.genesis_validators_root);
+        }
+        "root" => {
+            if let Some(state_id) = args.state_id {
+                let state = parse_state_id(state_id);
+                let out: Root = client.get_state_root(state).await.unwrap();
+                println!("{out:?}");
+            } else {
+                println!("no state-id provided");
+            }
+        }
+        "fork" => {
+            if let Some(state_id) = args.state_id {
+                let state = parse_state_id(state_id);
+                let out: Fork = client.get_fork(state).await.unwrap();
+                println!("{out:?}");
+            } else {
+                println!("no state-id provided");
+            }
+        }
+
+        _ => println!("endpoint missing or invalid - please provide valid endpoint"),
+    }
+}
+
+pub fn parse_state_id(state_id: String) -> StateId {
```

Comment:
we can leverage the existing code and libraries we rely on to avoid having to write out this parsing code ourselves

there are a few options:

we can use the `clap::value_parser` macro to specify exactly what is allowed here:

https://docs.rs/clap/latest/clap/_derive/_tutorial/index.html#validated-values

if we look at the docs (https://docs.rs/clap/latest/clap/macro.value_parser.html) then it says we can use this macro for any type that implements `From<String>` (and this family)

so one option would be to implement this trait on `StateId` and hook into the clap parsing machinery

beyond that, we may be able to just use the `serde::Deserialize` functionality in the `From<String>` implementation to not have to go case by case

it may take a little playing around with this to settle on something that is concise